### DISCUSSION TITLE
Invalid registry for transaction

### DISCUSSION THREAD

**Discussion by GuyHeylens** [project-origin/registry#131] [Q&A]
Hello, so we had some issues connecting to our Project Origin setup in Azure K8S from our development laptops. 

It looks like from another windows machine, I can sent something to our Project Origin setup. 
Unfortunately I get an "Invalid registry for transaction" .

Looking at the Example I see that in the args, you can pass 2 registries, 1 for consumption and 1 for production.
Do we need 2 instances of Project Origin or can we also run 1? 
<img width="1818" alt="image" src="https://github.com/project-origin/registry/assets/5173255/d2e8e2e8-0fcd-48aa-b087-99ff76b3253f">

After looking in the logs of Azure K8S I saw that the issue is in TransactionProcessor.cs on line 48:
![image](https://github.com/project-origin/registry/assets/5173255/b6452dcf-2704-4676-9db7-e76641c0fda5)

My Azure setup (values.yaml):

![values-registry-part](https://github.com/project-origin/registry/assets/5173255/d1e11eba-bc19-4540-9e00-aae106812cae)

My current WithoutWalletExample setup:
![image](https://github.com/project-origin/registry/assets/5173255/fb118edf-2e58-4812-aa3f-907b0bfaf158)

**Comment by MartinSchmidt** [comment]
There are 2 containers started, the verifier (for electricity specifically, and it uses the list you are refering to of registries) and the registry itself.
The registry itself currently gets the name from the name of the helm install (name) so one are actually required to install the chart with the same name as the registry, in your example

```shell
Helm install elia-registry ... 
```

I can already see this is not so great, requiring those to be the same. I will fasttrack and update to the chart, and enable an extra option to override it. 

I will fix it immediately.

**Comment by MartinSchmidt** [comment]
@GuyHeylens  I have fixed the issue in the latest release [v0.2.1](https://github.com/project-origin/registry/releases/tag/v0.2.1) give it a few moments to build the charts and containers.

And add the following to your values file at the root level

```yaml
registryName: elia-registry
```

Let me know if this resolved your issues :)
